index.js

```diff
@@ -132,7 +132,7 @@ module.exports = class ProtomuxRPC extends EventEmitter {
 
     if (request.timeout) clearTimeout(request.timeout)
 
-    if (error) request.reject(errors.REQUEST_ERROR(error))
+    if (error) request.reject(errors.REQUEST_ERROR('Request error', error))
```

Comment:
Sanity checking: is this change intended? Now all REQUEST_ERRORs show up with message 'request error', instead of the more specific message from before, and we need to check the `cause` field to get the actual error message. See the solo-ed test for an illustration of the behaviour.

I didn't know the `cause` option before, but from what I read its intention is to reference a more general error when throwing a more specific one, and we seem to be using it for the opposite purpose

> It [cause] is used when catching and re-throwing an error with a more-specific or useful error message in order to still have access to the original error.

Reply:
That's intended, yes. We can definitely improve the error message as those aren't part of the public API, unlike the error codes. The `cause` property is useful for layered error handling; `protomux-rpc` defines the error codes it may throw and for some of those errors callers may inspect `err.cause` for the underlying error.

```js
try {
  await rpc.request('some-command')
} catch (e) {
  if (e.code !== 'REQUEST_ERROR') throw err
  switch (e.cause.code) {
    // Handle the underlying error
  }
}

Reply:
What we _should_ do is remove all plain errors and use structured errors everywhere, including for the unknown method case.

Reply:
Okay, got it, the pattern makes sense.

What do you think of changing the err message back to what it was? Then all the old tests pass as before too. So it becomes:
```
if (error) request.reject(errors.REQUEST_ERROR(error.message, error))
```

Reply:
Happy to clean up the errors, either in this PR or a follow-up one

Reply:
We should do this: https://github.com/holepunchto/protomux-rpc/pull/11#pullrequestreview-2581346241